♻️ Feature/optimizations

topscoder opened the pull request:
Ehmmmm ... 🙉

* Introducing a new interactive installer to download & install Freqtrade (optional) and MoniGoMani HyperStrategy (optional)
* Refactored the logging functionality to log to file/stdout
* Rewrote most shell commands to use Python implementations for better cross OS compatibility
* Refactored `mgm-hurry`. Results in SoC -> more cleaner modules in `user_data/mgm_tools/mgm_hurry`
* Added unit tests for each of these modules
* Implemented mock objects for more pure unit testing
* Loads of polishing, cleaning, fixing
* Development: Implemented `pipenv` to run all commands in a clean and enclosed environment
* Development: Implemented `pre-commit` to run all checks and tests prior to committing code
* Development: GitHub Actions now always run: 
*  - Code Analysis *pylint flake8*
*  - Unit Tests *pytest*

.... 🙉

topscoder committed:
Generate report for each artifact (eg. python version)

topscoder committed:
Ignore testreport files

topscoder committed:
Also ignore test-results*.xml files

topscoder committed:
Added basic CONTRIBUTING document

topscoder committed:
Create codeql-analysis.yml

topscoder committed:
Merge branch 'feature/gh-action-ci' of https://github.com/topscoder/MoniGoMani into feature/gh-action-ci

topscoder committed:
Add pre-commit as dev-requirement

topscoder committed:
Fix imports

topscoder committed:
Fugbixing

topscoder committed:
Disabled storing test results caused by a bug

topscoder committed:
Won't get to sleep before this annoying issue is fixed 🙁

topscoder committed:
Add check-case-conflict

topscoder committed:
Being more specific in pre-commit rules

topscoder committed:
Being more specific

topscoder committed:
Being more specific

topscoder committed:
Being more specific

topscoder committed:
Reordered requirements-dev.txt contents

topscoder committed:
From double quotes " to single quotes '
Removed trailing whitespaces

topscoder committed:
Removed too many blank lines

topscoder committed:
Removed too many blank lines

topscoder committed:
Tweaking pytest and flake8 settings

topscoder committed:
Fixing action runners

topscoder committed:
Added CONTRIBUTING.md file

topscoder committed:
Split github workflows to remain a nice overview

topscoder committed:
Merge branch 'topscoder-feature/gh-action-ci' into development

topscoder committed:
Fixing some strings in mgm-hurry and add pre-commit example

topscoder committed:
Merge branch 'feature/gh-action-ci' into feature/add-tests-clean-code

topscoder committed:
Enable some additional pre-commit hooks and add missing encoding pragma lines

topscoder committed:
Do not enable pretty-format-json atm

topscoder committed:
Add flake8 style checker as pre-commit hook

topscoder committed:
Ignore junit-cache files in repo

topscoder committed:
Enable flake8 linting in vscode

topscoder committed:
🏋️‍♂️  Refactor FreqtradeCli to comply w/ flake8

topscoder committed:
😶‍🌫️  Restructuring FreqtradeCli

topscoder committed:
Updating fix-encoding-pragma pre-commit hook

topscoder committed:
Add wrapper around _exec_cmd to prevent calling private method

topscoder committed:
Fix setting/getting install_type in freqtradecli

topscoder committed:
Refator according to flake8 rules

topscoder committed:
Ignore freqtrade installation folders to prevent conflicts

topscoder committed:
Exclude freqtrade tests, but include mgm-tests

topscoder committed:
Ignore separate files from Freqtrade

topscoder committed:
Fixed some double-quotes to single-quotes

topscoder committed:
Some refactorings based on linter and unit test results

topscoder committed:
Wont save exchange/telegram settings as .hurry is tracked in repo

topscoder committed:
Removed MGM Hurry todo file as all todos are moved to github issues

topscoder committed:
Describes the use of pre-commit hooks in this repo.

topscoder committed:
Add MoniGoManiConfig module including unit tests

topscoder committed:
Fixing quotes in MoniGoManiConfig module

topscoder committed:
Adding tests

topscoder committed:
Fix access to dict keys

topscoder committed:
Merge remote-tracking branch 'origin/feature/cleaner-code' into feature/add-tests-clean-code

topscoder committed:
Everything cosmetics

topscoder committed:
Added more logging info in freqtradecli.py

topscoder committed:
Separated config-related instructions into monigomaniconfig module. Also fixes bugs.

topscoder committed:
Fixes style issues

topscoder committed:
Cleaner code

topscoder committed:
Renamed requirements.txt files to avoid conflicts with freqtrade files

topscoder committed:
Merge branch 'feature/add-tests-clean-code' of https://github.com/topscoder/MoniGoMani into feature/add-tests-clean-code

topscoder committed:
Fix GH actions to use renamed requirements files

topscoder committed:
Fix typo

topscoder committed:
Add configuration to launch mgm-hurry up in debug mode

topscoder committed:
Bugfixes where config files were not copied as it should

topscoder committed:
Fixes copy'ing example config files to usable config files

topscoder committed:
Don't overwrite existing files at installing freqtrade. Aka don't overwrite existing MGM files.

topscoder committed:
Fix bug where corrupt json config could crash mgm-hurry

topscoder committed:
Fixing issue where console output - mgm hurry questions - were not showing always

topscoder committed:
Merge branch 'feature/add-tests-clean-code' into origin/feature/add-tests-clean-code

topscoder committed:
Added some VSCode settings for linting and testing

topscoder committed:
Merge branch 'origin/feature/add-tests-clean-code' into feature/optimizations

topscoder committed:
Refactored command execution with / logging

topscoder committed:
Refactor – move code to the responsible objects

topscoder committed:
Pylint -> Polishing FreqtradeCli.py

topscoder committed:
Fixing ft logger

topscoder committed:
Refactoring all the way

topscoder committed:
Added vscode launch command for mgm-hurry up (for debugging)

topscoder committed:
freqtrade installation defaults to docker to avoid issues

topscoder committed:
Improvements to monigomaniconfig.py

topscoder committed:
Improving process handler and showing it's output live

topscoder committed:
Remove empty log files

topscoder committed:
Rewrite of the entire mgm logging module. Cleaner and easier to use now.

topscoder committed:
Cleanup unused logging code :)

topscoder committed:
Fixed all unit tests!

topscoder committed:
Skip overwriting freqtrade/mgm should not exit the entire procedure

topscoder committed:
Mistakenly overwritten README.md with freqtrade one. So, back to MGM!

topscoder committed:
Fix logger type

topscoder committed:
Fix MoniGoManiLogger initialization

topscoder committed:
Added editorconfig with some basic settings.

Borrowed from https://github.com/commaai/openpilot/blob/master/.editorconfig

topscoder committed:
Tweaking pylint config

topscoder committed:
Tweaked pre-commit hooks for a better world

topscoder committed:
Default values in .hurry file

topscoder committed:
Merge branch 'feature/optimizations' of https://github.com/topscoder/MoniGoMani into feature/optimizations

topscoder committed:
Polishing code

topscoder committed:
Merge branch 'feature/optimizations' of https://github.com/topscoder/MoniGoMani into feature/optimizations

topscoder committed:
Updated pip packages and pre-commit hooks

topscoder committed:
Updated the pytest configuration

topscoder committed:
Fixed double quotes

topscoder committed:
Shortened log lines so we got more (important) log data at a glance

topscoder committed:
Added Pipfiles so the pip requirements and installed versions are tracked

topscoder committed:
Rewrote the procedures for cloning git repo, copying files and executing shell commands

topscoder committed:
Cleanup unused code

topscoder committed:
Refactoring like a king 🤴

topscoder committed:
Fix typo

topscoder committed:
Refactoring w/ soc

topscoder committed:
Use fixed versions in pre-commit config file.

Also skip *MoniGoManiHyperStrategy.py files from linting atm.

topscoder committed:
Fixing quotes.. doh :)

topscoder committed:
Ignore *PairList.py at linting atm

topscoder committed:
Ignore linting TotalOverallSignalImporanceCalculator.py atm

topscoder committed:
Cleanup unused imports

topscoder committed:
Fix reading config file

topscoder committed:
🧹 Cleanup

topscoder committed:
Added encoding pragmas to avoid future charset encoding issues

topscoder committed:
Fixing comments in launch.json

JSON doesn't support comments according official specs. But didnt want to delete these comments as it can be very useful to find more information about the possible attributes.

topscoder committed:
Ignore unneccesary lambda warning

topscoder committed:
Fixing comments in JSON

topscoder committed:
Fix comment issue

topscoder committed:
Add pytest-cov as dev-requirement

topscoder committed:
Only run MGM unit tests (in tests folder)

topscoder committed:
Switch from pip -r to pipenv for running GitHub actions

topscoder committed:
Fixing pytest commands for GH actions

topscoder committed:
Fixing pytest

topscoder committed:
Fixing pytest configuration

topscoder committed:
Cleanup code conform style guide

topscoder committed:
Use pipenv to run pytest in gh actions

topscoder committed:
Fix running pytest. c'mon!

topscoder committed:
Aligned pip-requirements in pipfile and reqs.txt

topscoder committed:
Updated CONTRIBUTING.md

topscoder committed:
Show an interactive spinner while downloading freqtrade

topscoder committed:
Refactorings

topscoder committed:
Updated CONTRIBUTING.md

topscoder committed:
fixme -> FIXME

topscoder committed:
Merge branch 'feature/optimizations' of https://github.com/topscoder/MoniGoMani into feature/optimizations

topscoder committed:
Almost rewrote the entire code to download a static pairlist.

But, now the most FIXME's are fixed, code is polished more and less complex.

topscoder committed:
Cleanup code and solve FIXME's

topscoder committed:
Do not track freqtrade.egginfo folder

topscoder committed:
Do not track freqtrade.egg-info folder

topscoder committed:
Fix writing static pairlist to pairlist.json

topscoder committed:
Finished code for downloading static pairlist in a more stable manner

topscoder committed:
Use the mgm config module to interact with mgm configuration settings

topscoder committed:
startup_candle_count multiplier removed. Fixes rikj000/monigomani#91

topscoder committed:
Set 30m as default ticker instead of 1h

topscoder committed:
Added a bunch of run/debug configurations

topscoder committed:
Remove duplicate pragmas

topscoder committed:
Moved apply_best_results from mgm-hurry to monigomanicli.

Also rewritten the copy-process from shell-command to python-integrations

topscoder committed:
Store ho-results log file by default in user_data/ho_results/

topscoder committed:
Command bugfix for download_candle_data

topscoder committed:
Bugfix for running shell commands

topscoder committed:
mgm-launcher script to make first installation easy peasy

topscoder committed:
Default target folder for installation using launcher is monigomani/

topscoder committed:
📖 DOC: Updated MGM-Hurry.md docs to reflect the new mgm-launcher

topscoder committed:
Typo

topscoder committed:
Log debug messages as .debug(). not .info()

topscoder committed:
Launcher uses default requirements.txt to install pip packages

topscoder committed:
Move some functions from mgm-hurry to MoniGoManiConfig

topscoder committed:
Fix double quotes

topscoder committed:
Add developer notes to describe pipenv usage

topscoder committed:
Describe how to create ascii art module header

topscoder committed:
Code cleanup

topscoder committed:
🔶 Some polishing, cleanup, pydocs

topscoder committed:
Fix some style issues (flake8)

topscoder committed:
🤖 TEST: Add pytest-mock as dev-dependency

topscoder committed:
🤖 TEST: Added unit tests for mgm cli install type

topscoder committed:
🤖 TEST: Created unit tests for monigomani_cli.installation_exists

topscoder committed:
🤖 TEST: Writing unit tests

topscoder committed:
🐛 FIX:  Unit tests

topscoder committed:
Merge branch 'feature/optimizations' of https://github.com/topscoder/MoniGoMani into feature/optimizations

topscoder committed:
Moving test files to another directory to avoid conflicts with ft.

topscoder committed:
Moving test files to another directory to avoid conflicts with ft.

topscoder committed:
Moving test files to another directory to avoid conflicts with ft.

topscoder committed:
Moving unit-test files to another directory to avoid possible conflicts

topscoder committed:
flake8 style fixes

topscoder committed:
new installer

topscoder committed:
flake8

topscoder committed:
Remove sole monigomani file

topscoder committed:
Cleaning up files after chat with Rikj000

We decided not to blend to repo's together, so this repo will life on it's own.

topscoder committed:
Installer now makes installing freqtrade and mgm optional.

topscoder committed:
mgm-launcher is replaced by installer.sh

topscoder committed:
Match types because shell is random weird.

1 = false
0 = true

c'mon!

topscoder committed:
Fugbix installer

topscoder committed:
Fugbix installer.sh

topscoder committed:
Fugbixing installer.sh

topscoder committed:
Move unit tests to their original folder again

topscoder committed:
installer arguments wip

topscoder committed:
👌 IMPROVE: support arguments for branch and commit hash at installer.sh

topscoder committed:
🤖 TEST: Polished some tests

topscoder committed:
Fix access to __mgm_logger private var

topscoder committed:
Patch patch

topscoder committed:
patch patch

topscoder committed:
volare

topscoder committed:
Fixing unit tests. Period.

topscoder committed:
Merge remote-tracking branch 'rikj000/development' into merging

topscoder committed:
Pylint wants charset coding line.

topscoder committed:
Merging feature/optimizations and origin/development by hand. WIP

topscoder committed:
Remove duplicate config keys

topscoder committed:
Merged _read_config into MoniGoManiConfig.py

topscoder committed:
Implemented entire logging functionality in MGMLogger

And MGMLogger extends the default Python Logging functionality. Because leet.

topscoder committed:
Pylint and tox.ini same ignore-lines for code guidelines

topscoder committed:
Fixing some code styling

topscoder committed:
Renaming some methods for better distinction

topscoder committed:
👌 IMPROVE: Added some clarification to installer-script.

topscoder committed:
Workaround for GitHub security issue while PR testing

topscoder committed:
Update installation instructions

topscoder committed:
switch from bash to sh to support colors in shell

topscoder committed:
Splitting GH workflows to fix GH actions at PR

topscoder committed:
Fix logging filehandlers

topscoder committed:
Implement mgmlogger as singleton

topscoder committed:
Fix unit testing singleton instance

Rikj000 commented on mgm-hurry:
```diff
@@ -244,15 +243,21 @@ class MGMHurry:
 
             if result.get('proceed') is False:
-                sys.exit(1)
+                self._logger.info(
```
Refactor to single line

Rikj000 replied:
:heavy_check_mark: Done

topscoder replied:
Closed by 5eb4c8245554c613876f0a129fed8cc9e1e55091

Rikj000 commented on mgm-hurry:
```diff
@@ -446,27 +448,17 @@ class MGMHurry:
-                cleaned_up = True
-
-        strategy_config_hyperopt_path = f'{self._basedir}/user_data/strategies/{strategy}.json'
-        if os.path.exists(strategy_config_hyperopt_path):
```
This check should not be removed!
Other strategies create a `StrategyName.json` file after HyperOpting and it must be possible to clear these with `mgm-hurry cleanup`

topscoder replied:
Referenced #107 to discuss and fix.

topscoder replied:
Fixed by #5eb4c8245554c613876f0a129fed8cc9e1e55091

Rikj000 replied:
:heavy_check_mark: Done

Rikj000 reviewed (changes requested):
Thank you so much for this PR! Can't wait to merge it all in! :smile: 

However there are still some minor remarks that should be applied +
I'd like to test this PR thoroughly myself before merging it into `development` :slightly_smiling_face:

Rikj000 committed:
:zap: `MoniGoManiLogger` Match format of other `mgm_hurry` module headers

Rikj000 commented on user_data/mgm_tools/mgm_hurry/MoniGoManiLogger.py:
```diff
@@ -0,0 +1,270 @@
+#                                                                                 __/ |  __/ |
+#                                                                                |___/  |___/
+#
+# --- ↑↓ Do not remove these libs ↑↓ -----------------------------------------------------------------------------------
```
:heavy_check_mark: Done

Rikj000 committed:
:zap: `FreqtradeCli` Removed trailing comma & shortened some lines

Rikj000 commented on user_data/mgm_tools/mgm_hurry/FreqtradeCli.py:
```diff
@@ -1,94 +1,215 @@
-    @property
-    def basedir(self):
-        return self.__basedir
+        if self.install_type is None:
```
:heavy_check_mark: Done

Rikj000 committed:
:zap: `mgm-hurry` Refactored single line

Rikj000 committed:
:zap: `MoniGoManiCli` Refactored single line

Rikj000 commented on user_data/mgm_tools/mgm_hurry/MoniGoManiCli.py:
```diff
@@ -1,113 +1,141 @@
-        if os.path.exists(f'{target_dir}/user_data/mgm-config-private.json') is False:
-            shutil.copyfile(f'{target_dir}/user_data/mgm-config-private.example.json',
-                            f'{target_dir}/user_data/mgm-config-private.json')
+            repo = Repo.clone_from(GIT_URL_MONIGOMANI,
```
:heavy_check_mark: Done

Rikj000 commented on installer.sh:
```diff
@@ -0,0 +1,332 @@
+FREQTRADE_BRANCH="develop"
+FREQTRADE_COMMIT=""
+
+MGM_REPO_URL="https://github.com/topscoder/MoniGoMani.git"
```
We'll hold off with changing this until right before merging to maintain easy test-ability.

topscoder replied:
Correct, that's the purpose.

Rikj000 replied:
Apart from this one I believe all remarks are implemented by now! :partying_face: 
Aka, time to start testing locally :smile:

Rikj000 replied:
:heavy_check_mark: Last one, done!! :partying_face:

Rikj000 committed:
:zap: `MoniGoManiConfig` updated default HO Loss to `MGM_WinRatioAndProfitRatioHyperOptLoss`

Rikj000 commented on user_data/mgm_tools/mgm_hurry/MoniGoManiConfig.py:
```diff
@@ -0,0 +1,402 @@
+                    'exchange': 'binance',
+                    'hyperopt': {
+                        'strategy': 'MoniGoManiHyperStrategy',
+                        'loss': 'WinRatioAndProfitRatioLoss',
```
:heavy_check_mark: Done

Rikj000 commented on user_data/mgm_tools/mgm_hurry/FreqtradeCli.py:
```diff
@@ -1,94 +1,215 @@
-        if self.__install_type == 'docker':
+        if self.install_type == 'docker':
+            self.cli_logger.debug(
+                'FreqtradeCli::installation_exists() succeeded because install_type is set to docker.',
```
:heavy_check_mark: Done

Rikj000 replied:
:heavy_check_mark: Done

Rikj000 committed:
:zap: `FreqtradeCli` Nested freqtrade installation found checks

Rikj000 reviewed (changes requested):
Review updated, already checked off half of the remarks!
But now I should hit the sack.. :sweat_smile:
Tomorrow there's another day :relaxed:

topscoder committed:
indentation should be 4 spaces. not 2.

topscoder commented on .pylintrc:
```diff
@@ -334,19 +253,26 @@ expected-line-ending-format=
 # String used as indentation unit. This is usually "    " (4 spaces) or "\t" (1
 # tab).
-indent-string='    '
+indent-string='  '
```
Should be 4 indeed. Fixed in #89d23d4cf326003b7b345e405ca08c70362c02ec

Rikj000 replied:
:heavy_check_mark: Done

topscoder committed:
👌 IMPROVE: isort added as dependency

topscoder commented on Pipfile:
```diff
@@ -0,0 +1,23 @@
+yaspin = "==2.0.0"
+pytest-mock = "*"
+
+[dev-packages]
```
True. Added `isort` as dependency (+ sorted the packages list already) :)
#0881c2255c0bec34489e52ad2ecba089a797c17c

Rikj000 replied:
:heavy_check_mark: Done

topscoder committed:
🐛 FIX: Revert changes to docker-compose to last version (was accident)

topscoder commented on docker-compose.yml:
```diff
@@ -2,15 +2,17 @@
 version: '3'
```
True. Reverted back to your last version #aa4fb5a6a42d543832b9055e619bfcec1ae2ea6b

Rikj000 replied:
:heavy_check_mark: Done

topscoder committed:
📖 DOC: Updated Docs-MGM-Hurry.md

topscoder committed:
👌 IMPROVE: Also cleanup HO intermedia files from other strategies (fixes #107)

topscoder committed:
👌 IMPROVE: Typos and code styling fixes

topscoder commented on Documentation/Docs-MGM-Hurry.md:
```diff
@@ -35,6 +35,81 @@
 
 MGM-Hurry is a command line tool to speedup & simplify the setup & usage of [Freqtrade](https://www.freqtrade.io/en/stable/) in combination with the [MoniGoMani](https://github.com/Rikj000/MoniGoMani) FrameWork & Strategy.
```
Fixed in 5eb4c8245554c613876f0a129fed8cc9e1e55091

Rikj000 replied:
:heavy_check_mark: Done

topscoder committed:
📖 DOC: Optimized example command in installer.sh

topscoder committed:
📖 DOC: Optimized example command in installer.sh

topscoder committed:
👌 IMPROVE: More stable installer.sh

topscoder committed:
Move "art" to dev-packages

Rikj000 committed:
:beetle: Re-applied BugFix folder - json file mismatches in Docker

Rikj000 committed:
:memo: Reworked `Docs-MGM-Hurry.md`

Removed double documentation sections + Added ToDo's + Switched `shell` & `bash` => `powershell` code blocks to bring more color in them when displayed on GitHub + Fixed typo

Rikj000 committed:
:fire: Delete IDE specific `.vscode/launch.json` file

Rikj000 committed:
:fire: Delete IDE specific `.vscode/settings.json` file

Rikj000 committed:
:zap: `.gitignore` VSCodium Folder

Rikj000 committed:
:memo: `Docs-MGM-Hurry.md` Removed indentation from `installer.sh` command codeblock

Rikj000 committed:
:fire: Delete `mgm-config.json`

`mgm-config.example.json` (already in the repo) should be used so users won't accidentally over-write their own `mgm-config.json` file when pulling in the latest changes

Rikj000 commented on user_data/mgm-config.json:
```diff
@@ -0,0 +1,259 @@
+{
```
:heavy_check_mark: Done

Rikj000 commented on .vscode/launch.json:
```diff
@@ -0,0 +1,1448 @@
+{
```
:heavy_check_mark: Done

Rikj000 commented on .vscode/settings.json:
```diff
@@ -1,3 +1,11 @@
 {
```
:heavy_check_mark: Done

Rikj000 commented on installer.sh:
```diff
@@ -0,0 +1,336 @@
+
+# Install software dependencies
+python3 -m pip install pipenv
+pipenv install -r https://github.com/topscoder/MoniGoMani/blob/feature/optimizations/Pipfile
```
Throws following error when `pipenv` is not in `$PATH`:
`installer.sh - line 326: pipenv: command not found`

However when I change the command to `python3 -m pipenv install -r https://...`,
then I'm greeted with another error:
![pipenv-error](https://user-images.githubusercontent.com/24852597/129799553-f4dc97f1-d9ea-4632-83b1-e184141803e2.png)

topscoder replied:
Made this procedure a lot more robust. Already pushed.

Rikj000 replied:
:heavy_check_mark: Done

Rikj000 commented on user_data/mgm_tools/mgm_hurry/FreqtradeCli.py:
```diff
@@ -1,94 +1,209 @@
+#                        |_|
+
 import os
+from git import Repo
```
No `git` module is found when running `python3 ./mgm-hurry up`:
![no-git-module-error](https://user-images.githubusercontent.com/24852597/129800146-dddcb3f5-166b-41b2-9490-cfee52e8a663.png)

topscoder replied:
That's an environment issue because pipenv installed all modules in it's isolated environment (which is supposed to be).

Fixed now by proposing `pipenv run python3 ./mgm-hurry up` at the end of the installer.sh routine.

Note: `pipenv run ./mgm-hurry up` will in most cases also run fine, but adding python3 is probably more robust.

Rikj000 replied:
:heavy_check_mark: Done

Rikj000 reviewed (commented):
**Review update:**
Tested the `installer.sh` & `mgm-hurry up`, sadly both still appear to be having issues.
I'll do the rest of my testing after these 2 bugs have been resolved.

topscoder committed:
Merge branch 'feature/optimizations' of https://github.com/topscoder/MoniGoMani into feature/optimizations

topscoder committed:
🐛 FIX: installer.sh now also ensures installation of pip3, pipenv, pip packages

topscoder committed:
🐛 FIX: Make installing from requirements.txt more stable.

topscoder committed:
👌 IMPROVE: Make installer output more clear

topscoder committed:
📖 DOC: Polishing installer output

topscoder committed:
📖 DOC: Polishing installer output

topscoder committed:
📖 DOC: Gettings started command with pipenv prefix

topscoder committed:
🐛 FIX: Installing art package by default (not only dev)

topscoder committed:
👌 IMPROVE: Chaining shell commands to ensure it only proceeds without failures

topscoder committed:
🐛 FIX: Fix issue where mgm files were installed one directory level too deep. (strategies/strategies/*)

topscoder committed:
Removed weird ♂️ characters

topscoder committed:
Removed weird ♂️ characters

topscoder committed:
Some textual improvements

Rikj000 committed:
:zap: `MoniGoManiCli` Shorten error line

topscoder committed:
👌 IMPROVE: Blending installer and mgm-hurry to enforce install procedure

topscoder committed:
Merge branch 'feature/optimizations' of https://github.com/topscoder/MoniGoMani into feature/optimizations

topscoder committed:
Remove pipenv requirement in installer

topscoder committed:
👌 IMPROVE: Being consistent in Freqtrade naming, spinner allows for interactive input.

topscoder committed:
🐛 FIX: Fix printing error messages in loading spinner

Rikj000 reviewed (commented):
**Review Update:**
@topscoder it's looking all good to me now mate, amazing work truly! :partying_face: 
Just finished up a flawless install & it was a breeze to go through which made me smile with a HyperOpting machine at the end of it! :heart_eyes:

If you agree then I think we can finally merge it in! :tada:

Rikj000 committed:
:zap: `installer.sh` - Switched to Official MoniGoMani Repo + Recommended Freqtrade commit

Rikj000 committed:
:memo: Docs - Updated `installer.sh` command for Official MoniGoMani Repo

Rikj000 reviewed (approved):
Looking good to me! :smile: Let's finally merge into `development`! :tada:
